typescript-carousel/main.ts

```diff
@@ -0,0 +1,104 @@
+const images: string[] = [
+  './images/001.png',
+  './images/004.png',
+  './images/007.png',
+  './images/025.png',
+  './images/039.png',
+];
+
+let currNdx: number = 0;
+
+const $displayImage = document.querySelector(
+  '.display-image'
+) as HTMLImageElement;
+if (!$displayImage) throw new Error('.display-image query failed!');
+
+$displayImage.setAttribute('src', images[currNdx]);
+
+const $carouselContainer = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+if (!$carouselContainer) throw new Error('.carousel-container query failed!');
+
+const $leftArrow = document.querySelector('.fa-chevron-left');
+if (!$leftArrow) throw new Error('.fa-chevron-left query failed!');
+const $rightArrow = document.querySelector('.fa-chevron-right');
+if (!$rightArrow) throw new Error('.fa-chevron-right query failed!');
+
+const $circles = document.querySelectorAll('.fa-circle');
+if (!$circles) throw new Error('.fa-circles query failed!');
+console.log($circles);
+
+$carouselContainer.addEventListener('click', function (event: Event) {
+  console.log(event.target);
+  if (event.target === $leftArrow) {
+    if (currNdx >= 1) {
+      clearCircles();
+      setNdxImageAndDot(currNdx - 1);
+    } else if (currNdx === 0) {
+      clearCircles();
+      setNdxImageAndDot(4);
+    }
+  }
+
+  if (event.target === $rightArrow) {
+    if (currNdx <= 3) {
+      clearCircles();
+      setNdxImageAndDot(currNdx + 1);
+    } else if (currNdx === 4) {
+      clearCircles();
+      setNdxImageAndDot(0);
+    }
+  }
+
+  if (event.target === $circles[0]) {
+    clearCircles();
+    setNdxImageAndDot(0);
+  }
+  if (event.target === $circles[1]) {
+    clearCircles();
+    setNdxImageAndDot(1);
+  }
+  if (event.target === $circles[2]) {
+    clearCircles();
+    setNdxImageAndDot(2);
+  }
+  if (event.target === $circles[3]) {
+    clearCircles();
+    setNdxImageAndDot(3);
+  }
+  if (event.target === $circles[4]) {
+    clearCircles();
+    setNdxImageAndDot(4);
+  }
```

Comment:
There is a much better way to do this than to write a conditional for each item. Instead you should give your circles an attribute you can use to make changes rather than checking each one like this.

Reply:
can I used a switch statement instead? I'm changing the circles by using .classList.add in my setNdxImageAndDot() function isn't that similar to setting an attribute?